src/PreconInfo.tsx

```diff
@@ -68,25 +70,24 @@ export default function PreconForm() {
                     <FormControl>
                         <Select
                             id="set-select"
-                            value={selectedDeckState.deck.key}
-                            defaultValue={SELECT_A_DECK}
+                            value={selectedDeckState ?? SELECT_A_DECK}
                             fullWidth
-                            onChange={e => {
-                                // Select the deck from the JSON
-                                let selectedKey = Object.assign(new PreconDeck(), e.target.value);
-                                selectedKey.key = selectedKey.set.toLowerCase() + ":" + selectedKey.name.toLowerCase();
-                                let selectedDeck = selectPreconDeck(decks, selectedKey.key);
-                                setSelectedDeck({deck: selectedDeck});
-                                let partner: string = "";
-                                let commander = selectedDeck.commanders[0];
-                                if (selectedDeck.commanders.length > 1) {
-                                    partner = selectedDeck.commanders[1];
-                                }
-                                // Set the deck form
-                                setForm({ ...form, commander: commander, partner: partner, decklist: selectedDeck.decklist.join("\n") })
+                            onChange={(e: React.ChangeEvent<{ value: unknown }>) => {
+                                // Since MUI Select in not a real select element you will need to cast e.target.value
+                                // using as Type and type the handler as React.ChangeEvent<{ value: unknown }>
+                                // See https://stackoverflow.com/a/58676067/1053961
+                                let deck = e.target.value as PreconDeck;
+                                setForm({
+                                    ...form,
+                                    commander: deck.commanders[0],
+                                    partner: deck.commanders[1],
```

Comment:
Often the commanders array only has a single element. 
This will trigger an out of index error right? Also in JS?

Reply:
Nvm I tested this and it just returns an undefined, crazy language